index.d.ts

```diff
@@ -14,4 +14,4 @@ export type InitOptions = {
 }
 declare function init(options?: InitOptions): void
 
-export default init
+export = init
```

Comment:
this won't compile: you can't have both `export type InitOptions` and `export = ` in the same file:
```
TS2309: An export assignment cannot be used in a module with other exported elements.
```

A solution is to wrap the definitions you want to export into a namespace with the same name as what you export via `export =`:
```
declare namespace init {
   // your types here
}

declare function init (...)

export = init
```